src/super_context.ml

```diff
@@ -552,40 +551,240 @@ module Scope_key = struct
     sprintf "%s@%s" key (Dune_project.Name.encode scope)
 end
 
-module Action = struct
-  open Build.O
-  module U = Action.Unexpanded
+type targets =
+  | Static of Path.t list
+  | Infer
+  | Alias
+
+module Expander : sig
```

Comment:
Is there a way to split that to its own file or does it depend too much on what's before in the file?

Reply:
It depends on the `Super_context.t` itself. Once we have multi dir libs, we should be able to organize this better.